src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,102 @@
+<!--
+***********************************************************************************************
+Microsoft.Android.Sdk.Bindings.Gradle.targets
+
+This file contains MSBuild targets that support building and operating on Android Gradle projects.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.Gradle"              AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <_BuildAndroidGradleProjectsStamp>$(_AndroidStampDirectory)_BuildAndroidGradleProjects.stamp</_BuildAndroidGradleProjectsStamp>
+    <_AGPOutputCacheFile>agpoutputs.cache</_AGPOutputCacheFile>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <AndroidGradleProjectReference>
+      <Bind>true</Bind>
+      <Configuration>Release</Configuration>
+      <ModuleName></ModuleName>
+      <OutputPath>$(MSBuildProjectDirectory)/$(IntermediateOutputPath)gradle/%(Filename)/</OutputPath>
+      <Pack>true</Pack>
+      <ReferenceLibraryOutputs>true</ReferenceLibraryOutputs>
+      <Visible>false</Visible>
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs"
+        Condition=" '@(AndroidGradleProjectReference->Count())' != '0' ">
+    <ItemGroup>
+      <_AllGradleFiles Include="$([System.IO.Directory]::GetFiles('%(AndroidGradleProjectReference.FullPath)', '*.*', System.IO.SearchOption.AllDirectories))"
+          Condition= " Exists('%(AndroidGradleProjectReference.FullPath)') "/>
+      <_GradleInputs Include="@(_AllGradleFiles)"
+          Condition="  '%(Extension)' == '.java'
+                    or '%(Extension)' == '.kt'
+                    or '%(Extension)' == '.kts'
+                    or '%(Extension)' == '.gradle'
+                    or '%(Extension)' == '.properties'
+                    or '%(Extension)' == '.xml' " />
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="@(_AndroidMSBuildAllProjects)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="BuildAndroidGradleProjects"
```

Comment:
If this is a public target, then this should also be documented in `Documentation/docs-mobile/building-apps/build-targets.md`.

*Should* this be a public target?

Reply:
I've made this private.